### DISCUSSION TITLE
Customizing redirect location for MFR?

### DISCUSSION THREAD

**Discussion by fro0116** [TimMangan/MSIX-PackageSupportFramework#26] [Q&A]
Hi there! We have a few apps packaged using the FileRedirectionFixup in the old PSF. I only recently discovered this fork, and the MFR Fixup sounds like it could solve a ton of issues we've been having with the old Fixup (such as copy-on-access wasting tons of disk space)!

However, we're making heavy use of the redirectTargetBase option of FileRedirectionFixup to centralize package state for sync and backup, and looking through the docs I couldn't see an equivalent in MFR. Curious if I missed something or if this is just not supported?

Cheers!

**Comment by TimMangan** [comment]
Thank you for the feedback.

The MfrFixup does not have this feature.  

It seemed that nobody was using that feature in the FileRedirectionFixup so I did not include it in the rewrite.  I found that most customers needing to roam the settings were using a general third party tool to perform that redirection (Ivanti, FsLogix, Citrix User Profiles, etc).  When using those tools it is just a step to redo the app-specific rule to find the redirected files in the users AppData/Local/Packages folder. 

The good news is that the FileRedirectionFixup in this fork should be better than in the Microsoft source, but outside of this redirection control issue that you have, the Mfr is still way better.

Please let me know more about why you need the feature (especially where you are redirecting too) and I will consider it for the MFR, but no promises!

**Reply by fro0116** [reply]
Appreciate the quick response, Tim!

We've been redirecting to just a hard coded folder on the C: drive (would have preferred to redirect to somewhere in the user profile, but we could never get that working with the old fixup!). We then sync this folder using [Syncthing](https://syncthing.net/) across machines and to our NAS for backup.

We're a smaller shop and tend to prefer using free and open-source tools over proprietary vendors whenever possible, and this setup has worked pretty well for us outside of the issues with the FileRedirectionFixup itself. I realize it's a pretty niche use case and I'd totally understand if you don't end up supporting it.

Cheers!

**Reply by fro0116** [reply]
Hey Tim, quick update on this, we were able to get a new experimental sync setup working with the default AppData/Local/Packages folder, so we won't be needing the custom redirect location feature anymore after all! :)

We found a few issues along the way, so sent out a few PRs:

https://github.com/TimMangan/MSIX-PackageSupportFramework/pull/27
https://github.com/TimMangan/MSIX-PackageSupportFramework/pull/28

We're using a custom build with these changes so there's no rush, but would love to get these upstreamed whenever you have the chance to take a look. Cheers!
